tools/hw_decoder_bench.py

```diff
@@ -140,6 +143,78 @@ def create_input_tensor(batch_size, file_list):
     return np.stack(arrays)
 
 
+def non_image_preprocessing(raw_text):
+    return np.array([int(bytes(raw_text).decode('utf-8'))])
+
+
+@pipeline_def(batch_size=args.batch_size,
+              num_threads=args.num_threads,
+              device_id=args.device_id,
+              seed=0)
+def vit_pipeline(is_training=False, image_shape=(384, 384, 3), num_classes=1000):
+    files_paths = [os.path.join(args.images_dir, f) for f in os.listdir(
+        args.images_dir)]
+
+    img, clss = fn.readers.webdataset(
+        paths=files_paths,
+        index_paths=None,
+        ext=['jpg', 'cls'],
+        missing_component_behavior='error',
+        random_shuffle=False,
+        shard_id=0,
+        num_shards=1,
+        pad_last_batch=False if is_training else True,
+        name='webdataset_reader')
+
+    use_gpu = args.device == 'gpu'
+    labels = fn.python_function(clss, function=non_image_preprocessing, num_outputs=1)
+    if use_gpu:
+        labels = labels.gpu()
+    labels = fn.one_hot(labels, num_classes=num_classes)
+
+    device = 'mixed' if use_gpu else 'cpu'
+    img = fn.decoders.image(img, device=device, output_type=types.RGB,
+                            hw_decoder_load=args.hw_load,
+                            preallocate_width_hint=args.width_hint,
+                            preallocate_height_hint=args.height_hint)
+
+    if is_training:
+        img = fn.random_resized_crop(img, size=image_shape[:-1])
+        img = fn.flip(img, depthwise=0, horizontal=fn.random.coin_flip())
+
+        # color jitter
+        brightness = fn.random.uniform(range=[0.6, 1.4])
+        contrast = fn.random.uniform(range=[0.6, 1.4])
+        saturation = fn.random.uniform(range=[0.6, 1.4])
+        hue = fn.random.uniform(range=[0.9, 1.1])
+        img = fn.color_twist(
+            img,
+            brightness=brightness,
+            contrast=contrast,
+            hue=hue,
+            saturation=saturation)
+
+        # auto-augment
+        # `shape` controls the magnitude of the translation operations
+        img = auto_augment.auto_augment_image_net(img)
```

Comment:
The comment mentions `shape` argument, however there's no such thing in the code itself. Does it relate to the `img` argument? Like 
```python
auto_augment.auto_augment_image_net(shape=img)
```
?

Reply:
It was taken as it is from https://github.com/awolant/JAX-Toolbox/blob/f875b96d7d66916c6e4125f8315ddc197debf677/rosetta/rosetta/projects/vit/dali_update_test.py#L112.

Reply:
Anyway - removed comments.